services/githubService.js

```diff
@@ -0,0 +1,27 @@
+const logger = require('../utils/logger')
+const config = require('config')
+const { fetch } = require('../utils/fetch')
+const { fetchUser } = require('../models/users')
+
+/**
+ * Fetches the pull requests in Real-Dev-Squad by user using GitHub API
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const fetchPRsByUser = async (id) => {
+  try {
+    const { user } = await fetchUser(id)
+    const url = `${config.get('githubApi.baseUrl')}/search/issues?q=org:${config.get('githubApi.org')}+author:${user.github_id}+type:pr`
+    const { data } = await fetch(url)
+    return data
```

Comment:
NIT: What if we directly return fetch call, and destructure the data at call site?
@ankurnarkhede @swarajpure 

Anyway `TailCallOptimization` will be there, but at call site we are using await, which is not required because we are returning `data` directly, not a promise.
What's ideal in this case ?

Reply:
I have returned fetch call and destructured the data at function call

And about your second question, we need to use await at function call, otherwise `data.total_count` is undefined because `data` is not returned yet.